zebrad/tests/acceptance.rs

```diff
@@ -243,10 +253,90 @@ fn start_args() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn persistent_mode() -> Result<()> {
+    zebra_test::init();
+    let (tempdir, _guard) = tempdir(ConfigMode::Persistent)?;
+
+    let mut child = get_child(&["-v", "start"], &tempdir)?;
+
+    // Run the program and kill it at 1 second
+    std::thread::sleep(Duration::from_secs(1));
+    child.kill()?;
+    let output = child.wait_with_output()?;
+
+    // Make sure the command was killed
+    assert!(output.was_killed());
+
+    // Check that we have persistent sled database
+    let cache_dir = tempdir.join("state");
+    assert!(cache_dir.read_dir()?.count() > 0);
+
+    Ok(())
+}
+
+#[test]
+fn ephemeral_mode() -> Result<()> {
+    zebra_test::init();
+    let (tempdir, _guard) = tempdir(ConfigMode::Ephemeral)?;
+
+    // Any free argument is valid
+    let mut child = get_child(&["start", "argument"], &tempdir)?;
+    // Run the program and kill it at 1 second
+    std::thread::sleep(Duration::from_secs(1));
+    child.kill()?;
+    let output = child.wait_with_output()?;
+
+    // Make sure the command was killed
+    assert!(output.was_killed());
+
+    let cache_dir = tempdir.join("state");
+    assert!(!cache_dir.exists() || cache_dir.read_dir()?.count() > 0);
```

Comment:
Issue: When ephemeral is set, zebrad should not create the `cache_dir`.

```suggestion
    assert!(!cache_dir.exists());
```